scipy/ndimage/tests/test_morphology.py

```diff
@@ -2597,6 +2654,61 @@ def test_black_tophat04(self, xp):
         output = xp.empty_like(array, dtype=xp.float64)
         ndimage.black_tophat(array, structure=structure, output=output)
 
+    @skip_xp_backends("cupy",
+                      reasons=["these filters do not yet have axes support"],
+                      cpu_only=True,
+                      exceptions=['cupy', 'jax.numpy'])
```

Comment:
Please, not this pattern again :-).
Now that `skip_xp_backends` decorators can be stacked, we can express things is a less confusing way. So these tests need to skip all non-cpu backends, and additionally you skip cupy and jax.numpy on CPU, is this correct?
Then,
```
@skip_xp_backends(cpu_only)
@skip_xp_backends("cupy", reason="these filters do not yet have axes support in CuPy")
@skip_xp_backends("jax.numpy", reason="these filters do not yet have axes support in JAX.numpy")
```

Reply:
Thanks, I will update it